storage/transfermanager/downloader.go

```diff
@@ -546,11 +561,29 @@ type DownloadDirectoryInput struct {
 	// for syntax details. Optional.
 	MatchGlob string
 
-	// Callback will run after each object in the directory as selected by the
-	// provided filters is finished downloading. It must be set if and only if
-	// the [WithCallbacks] option is set.
-	// WaitAndClose will wait for all callbacks to finish.
-	Callback func(*DownloadOutput)
+	// Callback will run after all the objects in the directory as selected by
+	// the provided filters are finished downloading.
+	// It must be set if and only if the [WithCallbacks] option is set.
+	Callback func([]DownloadOutput)
+
+	// OnObjectDownload will run after every finished object download.
+	// It can only be set if the Downloader has the [WithCallbacks] option set.
```

Comment:
I think we could probably loosen this requirement? A user could want to run in sync mode and still use this to track/log progress.

Reply:
Done.